core/tests/protractor_utils/AdminPage.js

```diff
@@ -363,11 +363,20 @@ var AdminPage = function() {
   };
 
   this.expectUsernamesToMatch = async function(expectedUsernamesArray) {
+    if (expectedUsernamesArray.length === 0) {
+      await waitFor.textToBePresentInElement(
+        statusMessage, 'No results.',
```

Comment:
why `expectUsernamesToMatch` has to check the statusMessage? (Sorry, I'm unable to understand the check here!)

Also, why expectUsernamesToMatch should throw `'Toast with "No results." taking too long to appear'`?

Reply:
When we look up a username, it takes time for the admin page time to report those usernames. The toast tells us that the usernames have appeared. I put this in the `expectUsernamesToMatch` to keep the waitFor close to the code that accesses the usernames.

If no usernames are found, the toast shows `No results.`

Reply:
Should we handle this in the viewRolesbyUsername function i.e, the viewRolesbyUsername should end with wait for the toast to appear, that way the function will complete only after presenting the expected work? (It would be better for expectUsernamesToMatch to only validate values, right?)

Reply:
done